### PR TITLE
fix dissocPath for non-string types in path #2543

### PR DESCRIPTION
**Pull request by MilanLempera**
This change unifies behavior of `dissocPath` and `dissoc`. I only add tests and behavior by  https://github.com/ramda/ramda/blob/master/test/dissoc.js#L26

### CODE REVIEW

File: test/dissocPath.js [hunk 1]
```diff
@@ -60,4 +60,10 @@ describe('dissocPath', function() {
     eq(R.dissocPath([], {a: 1, b: 2}), {a: 1, b: 2});
   });
 
+  it('coerces non-string types', function() {
+    eq(R.dissocPath([42], {a: 1, b: 2, 42: 3}), {a: 1, b: 2});
+    eq(R.dissocPath([null], {a: 1, b: 2, 'null': 3}), {a: 1, b: 2});
+    eq(R.dissocPath([undefined], {a: 1, b: 2, undefined: 3}), {a: 1, b: 2});
```
Comment (CrossEye) [inline test/dissocPath.js]: I don't think we want to commit to this behavior on `null` and `undefined`.  We might one day want to insist that the user supplies an array of index types (strings, numbers, perhaps Symbols), so I would rather leave this behavior unspecified for the moment.

Can we remove those last two assertions?

**Comment by CrossEye** [comment]
This looks good, except that I think two of the tests are committing us to behavior better left unspecified for the moment.

**Comment by MilanLempera** [comment]
@CrossEye I have worked out the required changes, but I don't know why build job failed on node v0.12.18. 
I think fail is not caused by my change. Is there any way to run it again?

**Comment by CrossEye** [comment]
Sorry, I missed the previous message.  This looks good to me.  Anyone object?  @ramda/core?

**Comment by CrossEye** [comment]
Sorry, I have been little-focused on Ramda lately.  I will try to get back to it, release a version and work on the 1.0 and 2.0 roadmap.

**Comment by colinkahn** [comment]
@CrossEye no need to apologize! I'd be glad to help in some way if that's possible.

**Pull request merged**